### PR TITLE
Adapt Docker Proxy

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8399)
relates: https://github.com/demisto/demisto-sdk/pull/3975

## Description
https://confluence-dc.paloaltonetworks.com/pages/viewpage.action?pageId=121555211
Build Pipeline: https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/796097

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 1]
```diff
@@ -406,7 +406,7 @@
       - ${CI_PROJECT_DIR}/pipeline_jobs_folder/*
     when: always
   services:
-    - name: docker.art.code.pan.run/build-tools--image-dind:20.10.12-dind
+    - name: ${DOCKER_IO}/library/docker:20.10.12-dind
```
[the remark this answers is not in the document]
Comment (ydeutsch) [reply .gitlab/ci/.gitlab-ci.global.yml:409]: This isn't generic, this environment variable is specific for docker.io pull-through proxy

**Comment by ydeutsch** [comment]
Can you make sure there is also a default value for `DOCKER_IO` in the variables file? doesn't matter if it is `docker.io` or code.pan.run's

**Review by ydeutsch — APPROVED** [review]
Looks great, gets the job done (pun intended)

**Review by DeanArbel — APPROVED** [review]

**Comment by ydeutsch** [comment]
The authentication needed for using GCR an GAR is based on the job's Service Account which is dynamically discovered by gcloud using a metadata server
In order for this to actually happen gcloud can't have another active user since it will default to that user which will most likely not be authorized to access the repositories

**Comment by michal-dagan** [comment]
@DeanArbel The [PR](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/796097) needs force-merge:
- [NG errors](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/jobs/4543652) - not related issues
- [test-upload-flow ](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/796121)- same [issue](https://panw-global.slack.com/archives/C03A9TJ57FT/p1706706539661089)

**Pull request merged**